Py/DICOMSegmentationPlugin.py

```diff
@@ -155,130 +165,104 @@ def load(self,loadable):
-                                        typeModCode, typeModCodingScheme, typeModCodeMeaning,
-                                        regionCode, regionCodingScheme, regionCodeMeaning,
-                                        regionModCode, regionModCodingScheme, regionModCodeMeaning)
+    metaFileName = os.path.join(outputDir,"meta.json")
+
+    with open(metaFileName) as metaFile:
+      data = json.load(metaFile)
+      print ('number of segmentation files = ', numberOfSegments)
+      assert numberOfSegments == len(data["segmentAttributes"])
+      for segmentAttributes in data["segmentAttributes"]:
+        # TODO: only handles the first item of lists
+        segmentationColorNode.SetNumberOfColors(numberOfSegments + 1)
+        segmentationColorNode.SetColor(0, 'background', 0.0, 0.0, 0.0, 0.0)
+        for segment in segmentAttributes:
+          # load each of the segments' segmentations
+          # Initialize color and terminology from .info file
+          # See SEG2NRRD.cxx and EncodeSEG.cxx for how it's written.
+          # Format of the .info file (no leading spaces, labelNum, RGBColor, SegmentedPropertyCategory and
+          # SegmentedPropertyCategory are required):
+          # labelNum;RGB:R,G,B;SegmentedPropertyCategory:code,scheme,meaning;SegmentedPropertyType:code,scheme,meaning;SegmentedPropertyTypeModifier:code,scheme,meaning;AnatomicRegion:code,scheme,meaning;AnatomicRegionModifier:code,scheme,meaning
+          # R, G, B are 0-255 in file, but mapped to 0-1 for use in color node
+          # set defaults in case of missing fields, modifiers are optional
+
+          try:
+            rgb255 = segment["recommendedDisplayRGBValue"]
+            rgb = map(lambda c: float(c) / 255., rgb255)
+          except KeyError:
+            rgb = (0., 0., 0.)
+
+          segmentId = segment["LabelID"]
+
+          defaults = ['T-D0050', 'Tissue', 'SRT']
+          categoryCode, categoryCodingScheme, categoryCodeMeaning = \
+            self.getValuesFromCodeSequence(segment, "SegmentedPropertyCategoryCodeSequence", defaults)
+
+          typeCode, typeCodeMeaning, typeCodingScheme = \
+            self.getValuesFromCodeSequence(segment, "SegmentedPropertyTypeCodeSequence", defaults)
+
+          typeModCode, typeModCodingScheme, typeModCodeMeaning = \
+            self.getValuesFromCodeSequence(segment, "SegmentedPropertyTypeModifierCodeSequence")
+
+          anatomicRegionDefaults = ['T-D0010', 'SRT', 'Entire Body']
+          regionCode, regionCodingScheme, regionCodeMeaning = \
+            self.getValuesFromCodeSequence(segment, "AnatomicRegionSequence", anatomicRegionDefaults)
+
+          regionModCode, regionModCodingScheme, regionModCodeMeaning = \
+            self.getValuesFromCodeSequence(segment, "AnatomicRegionModifierSequence")
+
+          colorName = typeCodeMeaning
+          segmentationColorNode.SetColor(segmentId, colorName, *rgb)

```

Comment:
I guess I see your point  - you want to prepare it for @cpinter so that he can just swap the terminology part, right?